lib/tasks_private/vpc.rake

```diff
@@ -0,0 +1,168 @@
+      status = resource_controller.get_resource_instance(:id => cloud_db_id).result['state'] until status == 'active'
+    rescue => err
+      puts err
+    end
+
+    network_id = connection.create_vpc(:name => 'rake-network').result['id']
+    subnet_prototype = {
+      :vpc             => {
+        :id => network_id
+      },
+      :name            => 'rake-subnet',
+      :zone            => {
+        :name => 'us-south-1'
+      },
+      :ipv4_cidr_block => '10.240.0.0/24'
+    }
+    subnet_id = connection.create_subnet(:subnet_prototype => subnet_prototype).result['id']
+    volume_prototype = {
+      :profile  => {
+        :name => '5iops-tier'
+      },
+      :zone     => {
+        :name => 'us-south-1'
+      },
+      :name     => 'rake-vol',
+      :capacity => 10
+    }
+    volume_id = connection.create_volume(:volume_prototype => volume_prototype).result['id']
+
+    `ssh-keygen -f temp -t rsa -P ""`
+    ssh_key = `cat temp.pub`
+    auth_key_id = connection.create_key(:public_key => ssh_key, :name => 'rake-key').result['id']
+    `rm temp*`
+
+    security_group_id = connection.create_security_group(:vpc => {:id => network_id}, :name => 'rake-group').result['id']
+
+    instance_prototype = {
+      :primary_network_interface => {
+        :name            => 'eth0',
+        :subnet          => {
+          :id => subnet_id
+        },
+        :security_groups => [{
+          :id => security_group_id
+        }]
+      },
+      :name                      => 'rake-instance',
+      :zone                      => {
+        :name => 'us-south-1'
+      },
+      :vpc                       => {
+        :id => network_id
+      },
+      :profile                   => {
+        :name => "bx2-2x8"
+      },
+      :image                     => {
+        :id => 'r006-5e390f7b-0469-4ac1-b589-2fea307c1c5a'
+      },
+      :keys                      => [
+        {
+          :id => auth_key_id
+        }
+      ],
+      :volume_attachments        => [],
+      :boot_volume_attachment    => {
+        :volume => {
+          :name    => 'rake-instance-boot',
+          :profile => {
+            :name => 'general-purpose'
+          }
+        }
+      }
+    }
+    instance_id = connection.create_instance(:instance_prototype => instance_prototype).result['id']
+    status = connection.get_instance(:id => instance_id).result['status'] until status == 'running'
+
+    network_interface_id = connection.list_instance_network_interfaces(:instance_id => instance_id).result["network_interfaces"][0]["id"]
+    floating_ip_prototype = {:name => 'rake-floating-ip', :target => {:id => network_interface_id}}
+    floating_ip_id = connection.create_floating_ip(:floating_ip_prototype => floating_ip_prototype).result['id']
+    status = connection.get_floating_ip(:id => floating_ip_id).result['status'] until status == 'available'
+
+    # Generate VCRs
+    spec_file = Dir["#{spec_dir}/**/refresher_spec.rb"].first
```

Comment:
Do you want to only run the `vpc` specs (since this is vpc.rake)?  Maybe `spec_dir` should be `base_dir.join("models/manageiq/providers/ibm_cloud/vpc")` ?

Reply:
Oh you're just taking the first anyway, could you simply call out the full path to the `vpc/cloud_manager/refresher_spec.rb` then instead of the `Dir[].first`?